### PR TITLE
Use "security" section from autorest/modelerfour 

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Fix https://github.com/Azure/autorest.java/issues/1314

Code Change:
1. Map m4 output to CodeModel POJO
2. Add `SecurityInfo` property on `ServiceClient`, later will use this property to write to template
3. Update `ServiceClientMapper`  to map security related information to `SecurityInfo`. Note: we will first get security info from  CodeModel POJO `Security`, and then if `JavaSettings` has defined security related configuration, we will overwrite m4 security definition
4. Update templates(`ProtocolSampleTemplate`, `ProtocolTestBlankTemplate`, `ServiceClientBuilderTemplate`) to use `ServiceClient.SecurityInfo`
5. Add one vanilla test: https://github.com/Azure/autorest.java/tree/9abba471175727732704905c1c8fc49f0b4716cb/vanilla-tests/src/main/java/fixtures/securityinfo
5. Use https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Search/preview/2022-02-01-preview/geocoding.json to test locally. Generated builder looks like below. I also tested purview generation to make sure defining security info in JavaSetting works.

### CODE REVIEW

File: extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package com.azure.autorest.extension.base.model.codemodel;
```
Comment (weidongxu-microsoft) [inline extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java:1]: copyright

just config it in IDE

I will see if I got time to write a script to add them to existing classes and enable the checkstyle rule...

**Review by weidongxu-microsoft — COMMENTED** [review]
overall LGTM

File: extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java:1]: checkstyle rule merged

**Comment by weidongxu-microsoft** [comment]
Maps seems having good config in swagger
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Search/preview/2022-02-01-preview/geocoding.json

File: extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java (hunk 1 above)
Comment (haolingdong-msft) [reply extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/Scheme.java:1]: Thanks, will update

**Comment by haolingdong-msft** [comment]
> Maps seems having good config in swagger https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Search/preview/2022-02-01-preview/geocoding.json

Yes, I already use this json to test locally, it has both AzureKey and AzureToken. I will also add one case to our protocol test.

**Comment by weidongxu-microsoft** [comment]
I checked autorest.testserver does not have a swagger with AADToken.

You may want to add a very small swagger in https://github.com/Azure/autorest.java/tree/main/vanilla-tests/swagger for codegen of AADToken and AzureKey.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
> I checked that autorest.testserver does not have a swagger with AADToken.
> 
> You may add a very small swagger in https://github.com/Azure/autorest.java/tree/main/vanilla-tests/swagger with AADToken and AzureKey for test. Enable at https://github.com/Azure/autorest.java/blob/main/generate.bat#L47-L56 Generate code and commit to repo, this would be enough.

Agree, I will add

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**